src/Workspaces/Core/Portable/Workspace/Solution/ChecksumCollection.cs

```diff
@@ -58,7 +58,7 @@ public void AddAllTo(HashSet<Checksum> checksums)
         AssetPath assetPath,
         TextDocumentStates<TState> documentStates,
         HashSet<Checksum> searchingChecksumsLeft,
-        Dictionary<Checksum, object> result,
+        Func<Checksum, object, CancellationToken, ValueTask> onAssetFoundAsync,
```

Comment:
this allows finding to all be callback based, allowing us to write out the found assets to the oop<-->host stream as they are found.

Reply:
This also means we avoid a large dictoinary that has to be populated fully.